choice/src/Components/ChoiceStep.tsx

```diff
@@ -0,0 +1,157 @@
+import {Branch} from './Branch';
+import {
+  ActionGroup,
+  Button,
+  Card,
+  CardBody,
+  Flex,
+  FlexItem,
+  Form,
+  Stack,
+  StackItem
+} from '@patternfly/react-core';
+import {PlusCircleIcon} from '@patternfly/react-icons';
+import {useState} from 'react';
+
+type BranchProps = {
+  condition: string | null,
+  conditionSyntax: string,
+  identifier: string,
+  parameters: any[],
+  steps: any[]
+}
+
+type ChoiceStepProperties = {
+  branches?: BranchProps[];
+};
+
+export const ChoiceStep = (props: any) => {
+  const [step, setStep] = useState<ChoiceStepProperties>(props.step);
+  const [hasOtherwise, setHasOtherwise] = useState<boolean>(() => {
+    if (!step?.branches || step?.branches.length < 1) {
+    return false;
+    }
+    const maybeOtherwise = step?.branches[step.branches?.length - 1];
+    return maybeOtherwise.identifier === 'otherwise';
+  });
+
+  function updateKaoto() {
+    props.notifyKaoto('Choice step updated');
+    props.updateStep(step);
+  }
+
+  function setCondition(branch: BranchProps, condition: string, syntax: string) {
+    branch.condition = condition;
+    branch.identifier = condition;
+    branch.conditionSyntax = syntax;
+    setStep(step);
+  }
+
+  function handleDeleteBranch(branch: BranchProps) {
+    const index = step.branches?.indexOf(branch);
+    if (step.branches && index && index !== -1) {
+      const newBranches = step.branches.filter((_b, i) => i !== index);
+      step.branches = newBranches;
+      const isOtherwise = branch.identifier === 'otherwise';
+      if (isOtherwise) {
+        setHasOtherwise(false);
+      }
+      setStep(step);
+      props.notifyKaoto(`${isOtherwise ? 'Otherwise' : 'When: ' + (index+1)} deleted`);
+      props.updateStep(step);
+    }
+  }
+
+  function addWhen() {
+    const newWhen = {
+      condition: '',
+      conditionSyntax: 'SIMPLE',
+      identifier: '',
+      parameters: [],
+      steps: []
+    }
+    let index = -1;
+    if (!step.branches) {
+      step.branches = [newWhen];
+      index = 0;
+    } else {
+      index = hasOtherwise ? step.branches?.length - 1 : step.branches?.length;
+      step.branches.splice(index, 0, newWhen);
```

Comment:
Since we're not deleting any item (`0`), it's probably best we use `slice` instead of `splice` to not modify the `step.branches` array. I would actually advise against mutating `step` at all by just making a copy of it like: `const tempStep = {...step};` and then mutating that one instead, in which case `splice` would be fine!

Reply:
Ah thanks I was not aware of that mutation POV, will keep in my mind :+1: